### PR TITLE
CALSPEC standard stars list updated

### PR DESCRIPTION
**Pull request by debora-pe**
As titled

Companion Dev Suite PR [here](https://github.com/pypeit/PypeIt-development-suite/pull/317)

### CODE REVIEW

**Comment by debora-pe** [comment]
Run the tests. Only a vet test failed (I fixed in the [Dev Suite](https://github.com/pypeit/PypeIt/pull/1799)). The rest are failure not related to this and waiting for the NIRSPEC PR to clear
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  249 passed, 64 warnings in 342.06s (0:05:42) ---
--- PYTEST UNIT TESTS FAILED  3 failed, 131 passed, 168 warnings in 1648.35s (0:27:28) ---
--- PYTEST VET TESTS FAILED  1 failed, 60 passed, 97 warnings in 5174.56s (1:26:14) ---
--- PYPEIT DEVELOPMENT SUITE FAILED 1/237 TESTS  ---
Failed tests:
    keck_nirspec/LOW_NIRSPEC-1 pypeit
Skipped tests:
Testing Started at 2024-04-15T18:02:14.205806
Testing Completed at 2024-04-16T07:39:05.071855
Total Time: 13:36:50.866049
```

File: doc/releases/1.15.1dev.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ Instrument-specific Updates
 ---------------------------
 - Improvements for SOAR/Goodman
     - Added ``WAVMODE`` header keyword to ``configuration keys`` and
-     ``raw header cards``.
+      ``raw header cards``.
```
Comment (kbwestfall) [inline doc/releases/1.15.1dev.rst:32]: Also add a note about this PR?

File: doc/scripts/build_standardstbl_rst.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"""
```
Comment (kbwestfall) [inline doc/scripts/build_standardstbl_rst.py:1]: Nice!

File: doc/standards.rst [hunk 3]
```diff
@@ -15,58 +15,27 @@ defunct STScI ``calspec`` package.
...
-ltt9491_002.fits            LTT9491       23:19:35.00    -17:05:30.0
-feige110_005.fits           FEIGE110      23:19:58.40    -05:09:56.2
-========================== ============= ============== ==============
-
-Full list is available in ``pypeit/data/standards/calspec/README``.
-
-.. TODO: Can we just include that file here?
+The following table is an semi-complete list of the standard stars we are using from
+the STScI `CALSPEC calibration database
+<https://www.stsci.edu/hst/instrumentation/reference-data-for-calibration-and-tools/astronomical-catalogs/calspec>`__
```
Comment (kbwestfall) [inline doc/standards.rst]: Thanks for the updated text!  I went to this site and it provides references that users of the database should cite.  I would repeat that text here to make sure that pypeit users also cite those references.

File: pypeit/data/standards/calspec/README [hunk 4]
```diff
@@ -1,219 +1,577 @@
...
-standards is composed of IUE+Oke data only.
-
-CALSPEC also contains the ultraviolet to near-infrared absolute flux
-distribution of the Sun (filename: sun_reference_stis_001.fits) to 2.7
-microns. The solar reference spectrum combines absolute flux
-measurements from space and from the ground with a model spectrum for
-the near-infrared (Colina, Bohlin, & Castelli 1996). The newer
-absolute solar flux measurements of Thuillier (2003) may be a better
-choice for a solar reference spectrum from 1990-23975A.
+CALSPEC contains the composite stellar spectra that are flux standards on the HST system.
```
Comment (kbwestfall) [inline pypeit/data/standards/calspec/README:5]: Is this effectively a direct copy of the information at [the calspec site](https://www.stsci.edu/hst/instrumentation/reference-data-for-calibration-and-tools/astronomical-catalogs/calspec) ?  If so, I would add a note at the top of the doc that says this and add the date that it was copied.

File: pypeit/data/standards/calspec/calspec_info.txt [hunk 5]
```diff
@@ -1,34 +1,121 @@
...
-lds749b_stisnic_002.fits.gz  LDS749B    21:32:16.10 +00:15:15.0
-bd_28d4211_stis_001.fits     BD+284211  21:51:11.02 +28:51:50.4
-g93_48_004.fits              G93-48     21:52:25.38 +02:23:19.6
-bd_25d4655_002.fits          BD+254655  21:59:41.97 +26:25:57.4
-bd_17d4708_stisnic_002.fits  BD+174708  22:11:31.37 +18:05:34.2
-ngc7293_005.fits             NGC7293    22:29:38.55 -20:50:13.6
-ltt9491_002.fits             LTT9491    23:19:35.00 -17:05:30.0
-feige110_stisnic_002.fits    FEIGE110   23:19:58.40 -05:09:56.2
-g158100_flam.fits            G158100    00:33:54.46 -12:07:58.8
+# Semi-complete list of the calspec standards from:
```
Comment (kbwestfall) [inline pypeit/data/standards/calspec/calspec_info.txt:2]: Add a date at the top of the doc?

**Review by kbwestfall — APPROVED** [review]
Looks good!  I have a few minor doc suggestions, but I'm approving now.

File: doc/releases/1.15.1dev.rst (hunk 1 above)
Comment (debora-pe) [reply doc/releases/1.15.1dev.rst:32]: Done, thanks!

File: doc/standards.rst (hunk 3 above)
Comment (debora-pe) [reply doc/standards.rst]: Done!

File: pypeit/data/standards/calspec/README (hunk 4 above)
Comment (debora-pe) [reply pypeit/data/standards/calspec/README:5]: Done, thanks!

File: pypeit/data/standards/calspec/calspec_info.txt (hunk 5 above)
Comment (debora-pe) [reply pypeit/data/standards/calspec/calspec_info.txt:2]: Done

**Comment by debora-pe** [comment]
Doc updated

```
The HTML pages are in _build/html.

Build finished. The HTML pages are in _build/html.


========================
Sphinx build successful.
========================
```

**Review by jhennawi — APPROVED** [review]
Thanks!

**Pull request merged**